### DISCUSSION TITLE
Expose accessToken from accounts database

### DISCUSSION THREAD

**Discussion by tessamerrill** [nextauthjs/next-auth#1522] [Help]
**Summary of proposed feature**
I'm using the Discord OAuth sign in, and would like to interface with the API using the [Discord-OAuth2](https://npmjs.org/package/discord-oauth2) package on NPM, but I can't get the `accessToken` that's in the `accounts` database.

**Purpose of proposed feature**
Allow people to interface with the APIs that they use with this package.

**Potential problems**
None that I can think of.

**Describe any alternatives you've considered**
A hacky method of using an ORM and getting the token by finding by some unique identifier, in my case, find the user in the `users` database by email, use their `_id` (using MongoDB) to find their account in `accounts`, then using the `accessToken` from that.

*Please indicate if you are willing and able to help implement the proposed feature.*
Most likely not.

**Comment by naillizard** [comment]
Take a look at TypeORM [query builder](https://typeorm.io/#/select-query-builder/joining-relations).

**Comment by zvshka** [comment]
+1

**Comment by stale** [comment]
Hi there! It looks like this issue hasn't had any activity for a while. It will be closed if no further activity occurs. If you think your issue is still relevant, feel free to comment on it to keep it open. (Read more at #912) Thanks!

**Comment by almeidx** [comment]
I'm facing the same problem. (just trying to make the bot not close the issue)

**Comment by balazsorban44** [comment]
not sure about the OPs problem. You can get anything out from the account object in both the signIn and the jwt callback. Check the documentation.

https://next-auth.js.org/configuration/callbacks

do I miss something?
